### PR TITLE
Alternate fix for #730

### PR DESCRIPTION
**Pull request by williamb1024**
Modify Win32FileNativeMethods.CreateFile to return SafeFileHandle rather than IntPtr and adjust callsite to ensure the returned handle is not leaked.

Issue #730

### CODE REVIEW

**Comment by 304NotModified** [comment]
Thanks! 

Can you check the AppVeyor error? 

> NLog.UnitTests.Targets.Wrappers.AsyncTargetWrapperTests.AsyncTargetWrapperExceptionTest [FAIL]

**Comment by williamb1024** [comment]
The failing test isn't related to the changes. It looks to be a race condition within the test with the processing timer. The test does not fail when with the changes locally (so far anyway). I'm still playing with the test a bit to see if I can figure out exactly where the race is.

**Comment by 304NotModified** [comment]
Related #747

**Comment by 304NotModified** [comment]
This really looks nice :+1: 

But I have to admit I cannot oversee the impact of this change. What's your opinion? Is it safe to include in 4.0.1?

**Comment by williamb1024** [comment]
It's really just a "safer" version of the previous code, there should be no functionality change other than eliminating the overflow exception and ensuring that the native file handle is always closed in the case of an error. In the previous code it was possible that the handle would be leaked until the process terminated. 

There were mutliple possible leaks previously:

1. If an exception occurred after the native CreateFile completed and before SafeFileHandle took ownership of the handle. This could be caused by aborting the thread or an out of memory error (amongst other things). This would leak the native handle until the process terminated.
2. After SafeFileHandle took ownership but before FileStream took ownership of the SafeFileHandle. In this case, the leak would last until SafeFileHandle was finalized (which may or may not occur).

So, to summarize, the only visible difference will be that a native handle is not leaked in error cases.

**Comment by 304NotModified** [comment]
Thanks for the explanation. It's clear you have great knowledge regarding these handler stuff. :+1: :smiley: 

I'm more a MVC/Web developer ;)
